File: Programmers/Level2/더 맵게/더 맵게.java
import java.util.*;
class Solution {
    public int solution(int[] scoville, int K) {
        int answer = 0;
        PriorityQueue<Integer> que = new PriorityQueue<>();

        for(int i=0; i<scoville.length; i++){
            que.add(scoville[i]);
        }
        while(!check(que, K)){
            int tmp =que.poll();
            que.add(tmp+que.poll()*2);
            answer++;
            if(que.peek() < K && que.size()==1){
                return -1;
            }
        }
        return answer;
    }
    boolean check(PriorityQueue que, int K){
        if((int)que.peek() >= K){
            return true;
        }
        return false;
    }
    
}

File: Programmers/Level2/더 맵게/더 맵게.py
import heapq
def solution(scoville, K):
    answer = 0
    heapq.heapify(scoville)
    
    while len(scoville) > 1 and scoville[0] < K :
        heapq.heappush(scoville, heapq.heappop(scoville)+heapq.heappop(scoville)*2)
        answer += 1
        
    if scoville[0] < K :
        return -1
    else:
        return answer
